mm/2s2h/Rando/Logic/Regions/East.cpp

```diff
@@ -363,12 +363,12 @@ static RegisterShipInitFunc initFunc([]() {
             CHECK(RC_SECRET_SHRINE_WART_CHEST, RANDO_EVENTS[RE_SECRET_SHRINE_WART]),
             CHECK(RC_SECRET_SHRINE_GARO_MASTER_CHEST, RANDO_EVENTS[RE_SECRET_SHRINE_GARO_MASTER]),
             CHECK(RC_SECRET_SHRINE_PIECE_OF_HEART_CHEST, RANDO_EVENTS[RE_SECRET_SHRINE_DINALFOS] && RANDO_EVENTS[RE_SECRET_SHRINE_WIZZROBE] && RANDO_EVENTS[RE_SECRET_SHRINE_WART] && RANDO_EVENTS[RE_SECRET_SHRINE_GARO_MASTER]),
-            CHECK(RC_SECRET_SHRINE_POT_04, CAN_BE_ZORA),
-            CHECK(RC_SECRET_SHRINE_POT_05, CAN_BE_ZORA),
-            CHECK(RC_SECRET_SHRINE_POT_06, CAN_BE_ZORA),
-            CHECK(RC_SECRET_SHRINE_POT_07, CAN_BE_ZORA),
-            CHECK(RC_SECRET_SHRINE_POT_08, CAN_BE_ZORA),
-            CHECK(RC_SECRET_SHRINE_POT_09, CAN_BE_ZORA),
+            CHECK(RC_SECRET_SHRINE_POT_04, CAN_BE_ZORA || CAN_USE_ABILITY(RI_ABILITY_SWIM)),
+            CHECK(RC_SECRET_SHRINE_POT_05, CAN_BE_ZORA || CAN_USE_ABILITY(RI_ABILITY_SWIM)),
+            CHECK(RC_SECRET_SHRINE_POT_06, CAN_BE_ZORA || CAN_USE_ABILITY(RI_ABILITY_SWIM)),
+            CHECK(RC_SECRET_SHRINE_POT_07, CAN_BE_ZORA || CAN_USE_ABILITY(RI_ABILITY_SWIM)),
+            CHECK(RC_SECRET_SHRINE_POT_08, CAN_BE_ZORA || CAN_USE_ABILITY(RI_ABILITY_SWIM)),
+            CHECK(RC_SECRET_SHRINE_POT_09, CAN_BE_ZORA || CAN_USE_ABILITY(RI_ABILITY_SWIM)),
```

Comment:
I think what was here before was insufficient. If you can break them from a distance, then even human Link can dive down to get them.
```suggestion
            CHECK(RC_SECRET_SHRINE_POT_04, CAN_USE_PROJECTILE && CAN_USE_ABILITY(RI_ABILITY_SWIM)),
            CHECK(RC_SECRET_SHRINE_POT_05, CAN_USE_PROJECTILE && CAN_USE_ABILITY(RI_ABILITY_SWIM)),
            CHECK(RC_SECRET_SHRINE_POT_06, CAN_USE_PROJECTILE && CAN_USE_ABILITY(RI_ABILITY_SWIM)),
            CHECK(RC_SECRET_SHRINE_POT_07, CAN_USE_PROJECTILE && CAN_USE_ABILITY(RI_ABILITY_SWIM)),
            CHECK(RC_SECRET_SHRINE_POT_08, CAN_USE_PROJECTILE && CAN_USE_ABILITY(RI_ABILITY_SWIM)),
            CHECK(RC_SECRET_SHRINE_POT_09, CAN_USE_PROJECTILE && CAN_USE_ABILITY(RI_ABILITY_SWIM)),
```